app/lib/nhs/immunisations_api.rb

```diff
@@ -264,5 +336,37 @@ def check_vaccination_record_for_create_or_update(vaccination_record)
         raise "Patient nhs number is missing: #{vaccination_record.id}"
       end
     end
+
+    def check_bundle_link_params(bundle, request_params)
+      link = bundle.link&.find { |l| l.relation == "self" }&.url
+
+      uri = URI(link)
+      bundle_params = URI.decode_www_form(uri.query).to_h
+
+      # TODO: There is currently a bug in the API where the `Bundle.link` value for `-immunization.target` is
+      #       incorrectly returned as `immunization.target` rather than `-immunization.target`. Matt Jarvis has
+      #       told me that this should be fixed in their next release (3) or possibly release 4, at which point we
+      #       can remove this logic.
+      tweaked_bundle_params =
+        bundle_params.transform_keys do |k|
+          k == "immunization.target" ? "-immunization.target" : k
+        end
+
+      unless tweaked_bundle_params == request_params ||
+               bundle_params == request_params
+        raise NHS::ImmunisationsAPI::BundleLinkParamsMismatch,
+              "Bundle link parameters do not match request parameters: #{tweaked_bundle_params} != #{request_params}"
+      end
+    end
+
+    def check_operation_outcome_entry(bundle)
+      operation_outcome_entry =
+        bundle.entry&.find { |e| e.resource.resourceType == "OperationOutcome" }
```

Comment:
```suggestion
        bundle.entry&.find { it.resource.resourceType == "OperationOutcome" }
```

I'm not sure `e` is any more descriptive than `it`. 😅